File: IMS/IMS.Presentation/DTOs/GetDTO/TicketDTO.cs
using Shared.Enums;

namespace IMS.Presentation.DTOs.GetDTO;

public class TicketDto
{
    public Guid Id { get; init; }
    public Guid BoardId { get; init; } 
    public required string Title { get; init; } 
    public required string Description { get; init; } 
    public TicketStatus Status { get; set; } = TicketStatus.Unassigned; 
    public DateTime DeadLine { get; init; }
}

File: Frontend/ims-frontend/src/entities/ims/dto/ticket_dto.ts
import { TicketStatus } from "../enums";

export interface TicketDto {
    id: string;
    boardId: string;
    title?: string | null;
    description?: string | null;
    status: TicketStatus;
    deadLine: string;
}

export interface CreateTicketDto {
    boardId: string;
    title?: string | null;
    description?: string | null;
    status: TicketStatus;
    deadLine: string;
}

export interface UpdateTicketDto {
    title?: string | null;
    description?: string | null;
    status?: TicketStatus;
    deadLine?: string;
}
